robot/basestation/app.py

```diff
@@ -1,168 +1,206 @@
 #!/usr/bin/env python3
-# This is the controller page of the Flask GUI
-# Flask is light-weight and modular so this is actually all we need to set up a simple HTML page
+"""This is the controller page of the Flask GUI
+Flask is light-weight and modular so this is actually all we need to set up a simple HTML page"""
 
+import re
+import os
+import subprocess
 import flask
+from flask import jsonify
 
 app = flask.Flask(__name__)
 
-# set the rover's IP
-# rover static ip (tenatative): 192.168.1.20
-ROVER_IP = "127.31.43.134"
+# get rover's IP from global bash environment
+rover_ip = os.environ["ROS_MASTER_URI"]
```

Comment:
I suggest we setup environment variables a little more safely if we are going to need them. My suggestion is this:

Install `direnv` which autoloads environment variables specified in a `.envrc` file. Check [here](https://direnv.net/) for installation and basic usage.

Then, we create two files in the root project:
- `.envrc`: stores the env variables line by line such as `TEST_VAR="test"`.
- `setupvenv`: a simple shell script that allows to be referenced using `source` to start the `activate` script in the `venv/bin` and also initialize the setuptools file and activate the `direnv` tool. Example:
```
# setupvenv
source venv/bin/activate
python setup.py develop
direnv allow
```

This will take care of environment variables for us and not pollute the global profile. 

Every time we need to work on the project we `cd` into the root, and simply activate the environment and the rest of the toolchain by simply executing:
```
source setupvenv
```